commands/directMessage.go

```diff
@@ -0,0 +1,19 @@
+package commands
+
+import (
+	"context"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+// dmCommands is invoked when receiving a direct message from a user. If they are in the
+// process of registering, they cannot execute any other commands that they may normally
+// be able to invoke, and must complete the registering flow first
+func dmCommands(ctx context.Context, s *discordgo.Session, m *discordgo.MessageCreate) {
+	if state, ok := registering[m.Author.ID]; ok {
```

Comment:
One thing that may arise here is in the case of the bot being restarted during the register progress. It wouldnt accept any dm commands in this case and might confuse the user. Maybe add a default case to be called if the user doesnt have a state entry.

Reply:
Maybe we wanna consider storing signup tokens in consul or smth, would give us a place to see pending registrations if something goes wrong. And also to survive restarts

Reply:
Yeah thats a good idea